atomicfu/build.gradle.kts

```diff
@@ -128,52 +134,18 @@ kotlin {
     linuxArm32Hfp()
 
     @OptIn(ExperimentalKotlinGradlePluginApi::class)
-    applyDefaultHierarchyTemplate {
-        group("native") {
-            group("nativeUnixLike") {
-                withLinux()
-                withApple()
-            }
-        }
-        group("androidNative32Bit") {
-            withAndroidNativeX86()
-            withCompilations { compilation ->
-                (compilation.target as? KotlinNativeTarget)?.konanTarget?.name == "android_arm32"
-            }
-        }
-        group("androidNative64Bit") {
-            withAndroidNativeArm64()
-            withAndroidNativeX64()
-        }
-
-    }
+    applyDefaultHierarchyTemplate() 
 
     sourceSets {
-        val nativeUnixLikeMain by getting {
-            kotlin.srcDir("src/nativeUnixLikeMain/kotlin")
-            dependsOn(nativeMain.get())
-        }
-
-        val androidNative32BitMain by getting {
-            kotlin.srcDir("src/androidNative32BitMain/kotlin")
-            dependsOn(nativeMain.get())
-        }
-
-        val androidNative64BitMain by getting {
-            kotlin.srcDir("src/androidNative64BitMain/kotlin")
-            dependsOn(nativeMain.get())
-        }
-
-        val androidNative32BitTest by getting {
-            kotlin.srcDir("src/androidNative32BitTest/kotlin")
-            dependsOn(nativeTest.get())
-        }
-
-        val androidNative64BitTest by getting {
-            kotlin.srcDir("src/androidNative64BitTest/kotlin")
-            dependsOn(nativeTest.get())
-        }
-
+        
+        val concurrentMain by getting {}
+        val concurrentTest by getting {}
+        val nativeMain by getting { dependsOn(concurrentMain) }
+        val nativeTest by getting { dependsOn(concurrentTest) }
+        
+        val nativeUnixLikeMain by creating { dependsOn(nativeMain) }
+        val appleMain by getting { dependsOn(nativeUnixLikeMain) }
+        val linuxMain by getting { dependsOn(nativeUnixLikeMain) }
```

Comment:
Minor: these lines can be moved to the `kotlin {` block where `concurrentMain` and `concurrentTest` are created, which makes a couple of lines here redundant and allows this whole `sourceSets` block to be removed.